providers/flagd/pkg/provider.go

```diff
@@ -21,7 +21,7 @@ const defaultLRUCacheSize int = 1000
 
 type Provider struct {
 	ctx                              context.Context
-	Service                          service.IService
+	service                          service.IService
```

Comment:
This shouldn't be exported just for the sake of testing. Refactored tests into this same package to allow manipulation of this field.